lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -475,4 +509,62 @@ spec = do
             >>= (expectResponseCode HTTP.status403)
 
         r <- quitStakePool ctx (p ^. #id) (w, "Incorrect Passphrase")
-        expectResponseCode HTTP.status501 r
+        expectResponseCode HTTP.status403 r
+
+    it "STAKE_POOLS_QUIT_02 - Cannot quit non-existant stake pool" $ \ctx -> do
+        -- enable when delegation in WalletDelegation is updated
+        w <- fixtureWallet ctx
+
+        -- Join a pool
+        (_, p:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+        joinStakePool ctx (p ^. #id) (w, fixturePassphrase) >>= flip verify
+            [ expectResponseCode HTTP.status200
+            , expectFieldEqual status Pending
+            , expectFieldEqual direction Outgoing
+            ]
+
+        -- Wait for the certificate to be inserted
+        eventually $ do
+            let ep = listTxEp w mempty
+            request @[ApiTransaction n] ctx ep Default Empty >>= flip verify
+                [ expectListItemFieldEqual 0 direction Outgoing
+                , expectListItemFieldEqual 0 status InLedger
+                ]
+
+        let pId = B8.replicate 64 '0'
+        let (Right addr) = fromHex pId
+        let poolIdAbsent = ApiT $ PoolId addr
+        r <- quitStakePool ctx poolIdAbsent (w, "Secure Passphrase")
+        expectResponseCode HTTP.status404 r
+        expectErrorMessage (errMsg404NoSuchPool (T.decodeUtf8 pId)) r
```

Comment:
**[change required]** 

1. Remove `fromHex` call 
2.  Use `toText pId` instead of `T.decodeUtf8 pId`

Reply:
done